sdk/core/azure-core/src/base64.cpp

```diff
@@ -313,10 +313,10 @@ static void Base64WriteIntAsFourBytes(char* destination, int32_t value)
   destination[0] = static_cast<uint8_t>(value & 0xFF);
 }
 
-std::string Base64Encode(const std::vector<uint8_t>& data)
+std::string Base64Encode(uint8_t const* const data, size_t length)
```

Comment:
aaaaaa, whyyyyy ? why so many const ? it is const squared ?

Reply:
in C++, `const` binds to the left (which is weird). So `const * const` means a constant pointer to a constant value. 

In this case, it's probably not strictly necessary.